pkg/executor/executor.go

```diff
@@ -0,0 +1,49 @@
+package executor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type Interface interface {
```

Comment:
According to [effective go](https://go.dev/doc/effective_go#interface-names), it is better to name it `Executor`.